### DISCUSSION TITLE
Please help on Rotate the View

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#1959] [Support Q&A] [answered]
I would like to rotate the view as shown on the picture. I could not manage to rotate so far. The picture below is from another application.

![image](https://user-images.githubusercontent.com/5169314/151887269-b8ca76a8-316b-40da-8f67-80fc622feb16.png)

**Comment by nebsar** [comment, marked as answer]
At last it worked :) But, why do we have to use this code: viewer->setCameraManipulator(NULL); in each loop? If use I it at the beginning, it does not work.

![image](https://user-images.githubusercontent.com/5169314/151908904-4fe143fb-6208-4df6-a2d0-932b0f05ce47.png)

**Reply by hswjb** [reply]
I had the same problem, how did you rotate it?

**Reply by chrisjaquet** [reply]
@nebsar, it looks like `osgViewer::Viewer` automatically creates a `TrackballManipulator` when you call `run()` if there is no camera manipulator assigned (which happens if you set it to `NULL` at the beginning).

@hswjb, I believe the "correct" way to do this is to create your own `CameraManipulator` that you can use to set the orientation of the camera. This is not as easy as it sounds, unfortunately, especially if you still want all the default `EarthManipulator` functionality or want to handle other input. You can take a look at some of the other Manipulator examples in the OpenSceneGraph code (maybe `osgGA::FlightManipulator` I am not sure).
